### PR TITLE
Pullupstream/v0.5.6

### PR DESCRIPTION
**Pull request by rachit77**
This PR fetches v0.5.6 changes of the ZK-EVM node

### CODE REVIEW

File: cmd/run.go [hunk 1]
```diff
@@ -122,20 +121,8 @@ func start(cliCtx *cli.Context) error {
 		log.Fatal(err)
 	}
 
-	st := newState(cliCtx.Context, c, l2ChainID, []state.ForkIDInterval{}, stateSqlDB, eventLog, needsExecutor, needsStateTree)
+	st, currentForkID := newState(cliCtx.Context, c, tmpEthMan, l2ChainID, stateSqlDB, eventLog, needsExecutor, needsStateTree, false)
```
Comment (arnaubennassar) [inline cmd/run.go]: you should create the real etherman right after the state, the use the real etherman everywhere after this step

File: config/cardonagenesis.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
+package config
+
+// CardonaNetworkConfigJSON is the hardcoded network configuration to be used for the official mainnet setup
+const CardonaNetworkConfigJSON = `
```
Comment (arnaubennassar) [inline config/cardonagenesis.go]: remove this file

File: config/environments/cardona/example.env [hunk 3]
```diff
@@ -0,0 +1,9 @@
+ZKEVM_NETWORK = "cardona"
```
Comment (arnaubennassar) [inline config/environments/cardona/example.env]: remove this file and other files from this directory (`config/environments`)

File: config/mainnetgenesis.go [hunk 4]
```diff
@@ -5,8 +5,9 @@ const MainnetNetworkConfigJSON = `
 {
 	"l1Config" : {
 		"chainId": 1,
-		"polygonZkEVMAddress": "0x5132A183E9F3CB7C848b0AAC5Ae0c4f0491B7aB2",
-		"polTokenAddress": "0x7D1AfA7B718fb893dB30A3aBc0Cfc608AaCfeBB0",
+		"polygonZkEVMAddress": "0x519E42c24163192Dca44CD3fBDCEBF6be9130987",
```
Comment (arnaubennassar) [inline config/mainnetgenesis.go]: remove

**Review by arnaubennassar — CHANGES_REQUESTED** [review]
Please update `docs/diff/diff.html`

**Comment by begmaroman** [comment]
Check linter issues please

File: cmd/run.go [hunk 5]
```diff
@@ -122,20 +121,8 @@ func start(cliCtx *cli.Context) error {
...
 
-	etherman, err := newEtherman(*c, st)
```
Comment (goran-ethernal) [inline cmd/run.go:127]: Is this safe to remove?

Comment (rachit77) [reply cmd/run.go:127]: > Is this safe to remove?

`etherman` instance has not been removed.
It is now being created at line 130

Comment (goran-ethernal) [reply cmd/run.go:127]: I see now. Thanks.

File: etherman/simulated.go [hunk 6]
```diff
@@ -204,9 +209,8 @@ func NewSimulatedEtherman(cfg Config, auth *bind.TransactOpts, daBackend dataAva
 		return nil, nil, common.Address{}, nil, err
 	}
 
-	client.Commit()
```
Comment (arnaubennassar) [inline etherman/simulated.go:207]: do not remove this line

**Review by arnaubennassar — CHANGES_REQUESTED** [review]

**Review by arnaubennassar — APPROVED** [review]

**Pull request merged**